overlord/configstate/configcore/snapshots.go

```diff
@@ -0,0 +1,54 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.automatic-snapshots.expiration"] = true
+}
+
+func validateAutomaticSnapshotsExpiration(tr config.Conf) error {
+	expirationStr, err := coreCfg(tr, "automatic-snapshots.expiration")
```

Comment:
my  first suggestion would be   "snapshots.auto-saves-expiration",  but I'm keen on suggestions from @degville or @chipaca.   I'm also not sure whether instead of "expiration"  we should use something that makes it positive about the time they are kept  like "lifetime", "shelf-life" , not sure ?

Reply:
from my non-technical perspective, I agree with `snapshots` for the config namespace - it's where I'd expect to find an option like this, as it would like be more discoverable grouped with other (potential) snapshot-related options.

for the setting name itself, it's a good question! I can see advantages to both - *expiration* makes it clear you're losing data. But I think, on balance, I agree with making it positive, as @pedronis suggests. It seems a little easier to mentally parse for me. Maybe with something like `snapshots.retention-period`?

Reply:
`snapshots.auto-saves-retention` (from IRC) seems to be the winner so far.
`expiration` is bad because the thing you set is a duration, not a date (and expiration would be a date)

Reply:
so we discussed further and converged on snapshots.automatic.retention , and use "no" as the special value to use if one doesn't want it

Reply:
Done